### ISSUE TITLE
Changing vaadin to vaadin-core dependency can break the application with 14.2

### ISSUE THREAD

**Issue by TatuLund** [vaadin/flow#8448]
Steps to reproduce

- Use Beverage Buddy test application
- Check that it has Vaadin version 14.2

Change the project to use

        <dependency>
            <groupId>com.vaadin</groupId>
            
            <artifactId>vaadin-core</artifactId>
            ...
        </dependency>

When application starts it is broken and you can find exception on console log

```
Uncaught DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "vaadin-overlay" has already been used with this registry
    at eval (webpack-internal:///../node_modules/@vaadin/vaadin-dialog/node_modules/@vaadin/vaadin-overlay/src/vaadin-overlay.js:1098:16)
    at Module.../node_modules/@vaadin/vaadin-dialog/node_modules/@vaadin/vaadin-overlay/src/vaadin-overlay.js (http://localhost:8080/VAADIN/build/vaadin-bundle-36969be0b59ccab111c7.cache.js:1872:1)
    at __webpack_require__ (http://localhost:8080/VAADIN/build/vaadin-bundle-36969be0b59ccab111c7.cache.js:20:30)
    at eval (webpack-internal:///../node_modules/@vaadin/vaadin-dialog/src/vaadin-dialog.js:5:102)
    at Module.../node_modules/@vaadin/vaadin-dialog/src/vaadin-dialog.js (http://localhost:8080/VAADIN/build/vaadin-bundle-36969be0b59ccab111c7.cache.js:1908:1)
    at __webpack_require__ (http://localhost:8080/VAADIN/build/vaadin-bundle-36969be0b59ccab111c7.cache.js:20:30)
    at eval (webpack-internal:///../node_modules/@vaadin/vaadin-dialog/theme/lumo/vaadin-dialog.js:3:79)
    at Module.../node_modules/@vaadin/vaadin-dialog/theme/lumo/vaadin-dialog.js (http://localhost:8080/VAADIN/build/vaadin-bundle-36969be0b59ccab111c7.cache.js:1932:1)
    at __webpack_require__ (http://localhost:8080/VAADIN/build/vaadin-bundle-36969be0b59ccab111c7.cache.js:20:30)
    at eval (webpack-internal:///../target/frontend/generated-flow-imports.js:36:108)
Collaps
```

![image](https://user-images.githubusercontent.com/14309836/83123995-43dd9380-a0de-11ea-91a8-c1288d74aa69.png)

Removal of node_modules, package-lock.json, ... and doing clean build does not help.

Apparently this is not happening with all applications.

**Comment by pleku** [comment]
The workaround should be to remove `node_modules` and `package-lock.json`.

Not clear why there would be a difference in the versions being resolved between `vaadin` and `vaadin-core`.

**Comment by TatuLund** [comment]
> The workaround should be to remove node_modules and package-lock.json.

It did not work in this case.

**Comment by pleku** [comment]
It seems that the issue is now resolved as the LATEST tag for vaadin-overlay has been updated https://www.npmjs.com/package/@vaadin/vaadin-overlay?activeTab=versions as it was pointing to `3.2.20`, kudos for @web-padawan noticing this.

Closing the issue as it seems that preventing this from happening should be done on the platform level and ⚙️ are in motion.

**Comment by mvysny** [comment]
So, is it fixed now, or do we need to wait for Vaadin 14.2.1?

**Comment by pleku** [comment]
@mvysny as said above it was fixed after the tag update. So no need to wait for anything.
